src/SFA.DAS.AssessorService.Web/Controllers/Apply/StandardController.cs

```diff
@@ -353,6 +381,35 @@ private IEnumerable<StandardVersionApplication> ApplyVersionStatuses(IEnumerable
                 }
             }
 
+            var withdrawals = previousWithdrawals.Where(x => x.StandardApplicationType == StandardApplicationTypes.VersionWithdrawal);
```

Comment:
I think there are some holes in the logic added to this method:
- It doesn't take into account the current version status, if a version is approved, and it previously had a withdrawal application, it's going to get overridden, in what cases should the withdrawal status be set? (ie not approved or in progress)
- You would only want the latest withdrawal applications for the versions of the standard, not all. ( ie withdraw, apply withdraw again )
- EPA changed could happen across multiple versions not just the latest so needs to be taken into account.

I think you would be better off altering the `MapUnapprovedVersionStatus` method, whereby you are looping through the versions and their state, pass in the previous withdrawal applications into the method, and then under the withdrawn status, apply a shorter version of the logic
- For the version that's withdrawn, find it's withdrawal application and it's withdrawal date.
- Change it's version status from Withdrawn to the opt in or 7 apply depending.

or something similiar to the above?

Reply:
Refactored - added `!= approved` status check

Reply:
`!= approved` has been corrected

Have tested this scenario and seems as if it works ok - unless theres a scenario I've missed,
I will catch up with Uma to see what hes getting in the test env